scripts/deploy.ts

```diff
@@ -1,20 +1,20 @@
-import { ethers } from "hardhat";
+import {ethers, upgrades} from "hardhat";
 
 async function main() {
 
-  const CryptoTip = await ethers.getContractFactory("CryptoTip");
-  const cryptoTip = await CryptoTip.deploy();
+    const CryptoTip = await ethers.getContractFactory("CryptoTip");
+    const cryptoTip = await upgrades.deployProxy(CryptoTip, {initializer: 'initialize'});
```

Comment:
Using proxy contracts is not something I know a lot about. I need to learn more.